src/core/server/execution_context/integration_tests/tracing.test.ts

```diff
@@ -0,0 +1,427 @@
+        .get(root, '/execution-context')
+        .set('x-opaque-id', myOpaqueId)
+        .expect(200);
+
+      const header = response.body['x-opaque-id'];
+      expect(header).toBe(myOpaqueId);
+    });
+
+    it('generated and attached to Elasticsearch unscoped client calls if not specifed', async () => {
+      const { http } = await root.setup();
+      const { createRouter } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        const { headers } = await context.core.elasticsearch.client.asInternalUser.ping();
+        return res.ok({ body: headers || {} });
+      });
+
+      await root.start();
+
+      const response = await kbnTestServer.request.get(root, '/execution-context').expect(200);
+
+      const header = response.body['x-opaque-id'];
+      expect(header).toEqual(expect.any(String));
+    });
+
+    it('generated and attached to Elasticsearch scoped client calls if not specifed', async () => {
+      const { http } = await root.setup();
+      const { createRouter } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        const { headers } = await context.core.elasticsearch.client.asCurrentUser.ping();
+        return res.ok({ body: headers || {} });
+      });
+
+      await root.start();
+
+      const response = await kbnTestServer.request.get(root, '/execution-context').expect(200);
+
+      const header = response.body['x-opaque-id'];
+      expect(header).toEqual(expect.any(String));
+    });
+
+    it('can be rewritten during Elasticsearch client call', async () => {
+      const { http } = await root.setup();
+      const { createRouter } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        const { headers } = await context.core.elasticsearch.client.asInternalUser.ping(
+          {},
+          {
+            opaqueId: 'new-opaque-id',
+          }
+        );
+        return res.ok({ body: headers || {} });
+      });
+
+      await root.start();
+
+      const myOpaqueId = 'my-opaque-id';
+      const response = await kbnTestServer.request
+        .get(root, '/execution-context')
+        .set('x-opaque-id', myOpaqueId)
+        .expect(200);
+
+      const header = response.body['x-opaque-id'];
+      expect(header).toBe('new-opaque-id');
+    });
+  });
+  describe('execution context', () => {
+    it('sets execution context for a request handler', async () => {
+      const { executionContext, http } = await root.setup();
+      const { createRouter } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        executionContext.set(parentContext);
+        await delay(100);
```

Comment:
This sort of concerns me. Dp we need this delay? Will this not work on endpoints without any async calls? Even if all endpoints that call Elasticsearch will have async calls, we may still want log correlation for endpoints without async calls.

Reply:
I added `delay` to show that `set` works for the async operations. I will add another test to prove the `set` works for sync operations as well.